persistence/persistence-deltaspike/src/main/java/uk/gov/justice/services/persistence/EntityManagerProducer.java

```diff
@@ -11,21 +13,23 @@
 /**
  * Producer of {:link EntityManager} for use with JPA (Delta-spike).
  */
+@ApplicationScoped
 public class EntityManagerProducer {
     private static final String UTC = "UTC";
 
     @PersistenceUnit
     EntityManagerFactory entityManagerFactory;
 
     @Produces
+    @RequestScoped
```

Comment:
Background: 
 Container Managed EntityManagerProducer as Application Scoped (1 for each war)
 Injected Entity Manager Factory (1 for each war) 
 Contextual (programatic) Entity manager (Request Scoped)   (1 for each MDB)
which creates request scoped PersistentContext and attaches a request scoped TransactionContext

Impact of Request Scope in terms of transaction context is active from the start to end of the MDB, yes so processing a message cound as a request scoped  transaction. 
Also transactional test (add and remove) passes now which was failing before using bean managed Persistence/transactions